app/models/income_statement.rb

```diff
@@ -119,4 +255,40 @@ def totals_query(transactions_scope:)
     def monetizable_currency
       family.currency
     end
+
+    def calculate_savings_rate(total_income, total_expenses)
+      return 0 if total_income.zero?
+      savings = total_income - total_expenses
+      rate = (savings / total_income.to_f) * 100
+      rate.round(2)
+    end
+
+    # Get previous period for comparison
+    def get_previous_period(period)
+      if period.is_a?(Period)
+        # For custom periods, create a period of same length ending right before this period starts
+        length = (period.end_date - period.start_date).to_i
+        Period.new(start_date: period.start_date - length.days - 1.day, end_date: period.start_date - 1.day)
+      else
+        # Default to previous month
+        current_month = Date.today.beginning_of_month..Date.today.end_of_month
+        previous_month = 1.month.ago.beginning_of_month..1.month.ago.end_of_month
+        Period.new(start_date: previous_month.begin, end_date: previous_month.end)
+      end
+    end
+
+    # Get a human-readable name for a period
+    def period_name(period)
+      if period == Period.current_month
+        "Current Month"
+      elsif period == Period.previous_month
+        "Previous Month"
+      elsif period == Period.year_to_date
+        "Year to Date"
+      elsif period == Period.previous_year
+        "Previous Year"
+      else
+        "#{format_date(period.start_date)} to #{format_date(period.end_date)}"
+      end
+    end
```

Comment:
When using the chat, I ran into an application error pretty much any time I asked about IncomeStatement items that had a period:

```
[ActiveJob] [ProcessAiResponseJob] [2909c435-6d15-40fe-b895-1519239cfd38] Error generating AI response: undefined method 'previous_month' for class Period
16:28:49 web.1    | [ActiveJob] [ProcessAiResponseJob] [2909c435-6d15-40fe-b895-1519239cfd38] /Users/zachgoll/Documents/apps/maybe/app/models/ai/financial_assistant.rb:520:in 'Ai::FinancialAssistant#get_period_from_param'
```